subprojects/base-services/src/main/java/org/gradle/internal/classloader/ClassLoaderUtils.java

```diff
@@ -80,22 +64,14 @@ public static void disableUrlConnectionCaching() {
         }
     }
 
-    public static JavaMethod<ClassLoader, Package[]> getPackagesMethod() {
-        return GET_PACKAGES_METHOD;
-    }
-
-    public static JavaMethod<ClassLoader, Package> getPackageMethod() {
-        return GET_PACKAGE_METHOD;
+    static ClassLoaderPackagesFetcher getClassLoaderPackagesFetcher() {
+        return CLASS_LOADER_PACKAGES_FETCHER;
```

Comment:
I'm wondering whether we should expose this interface or simply provide two static functions in `ClassLoaderUtils` that delegate.

Reply:
That's a great idea. We shouldn't expose the interfaces. Already changed to static methods.